extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeStreamExtractor.java

```diff
@@ -316,6 +318,65 @@ public String getSupportInfo() {
         }
     }
 
+    @Nonnull
+    @Override
+    public List<StreamSegment> getStreamSegments() throws ParsingException {
+        final List<StreamSegment> segments = new ArrayList<>();
+        final JsonObject segmentsJson;
+        try {
+            segmentsJson = fetchSubApiContent("chapters");
+        } catch (final IOException | ReCaptchaException e) {
+            throw new ParsingException("Could not get stream segments", e);
+        }
+        if (segmentsJson != null && segmentsJson.has("chapters")) {
+            final JsonArray segmentsArray = segmentsJson.getArray("chapters");
+            for (int i = 0; i < segmentsArray.size(); i++) {
+                final JsonObject segmentObject = segmentsArray.getObject(i);
+                segments.add(new StreamSegment(
+                        segmentObject.getString("title"),
+                        segmentObject.getInt("start")));
+            }
+        }
+
+        return segments;
+    }
+
+    @Nonnull
+    @Override
+    public List<Frameset> getFrames() throws ExtractionException {
+        final List<Frameset> framesets = new ArrayList<>();
+        final JsonObject storyboards;
+        try {
+            storyboards = fetchSubApiContent("storyboards");
+        } catch (final IOException | ReCaptchaException e) {
+            throw new ExtractionException("Could not get frames", e);
+        }
+        if (storyboards != null && storyboards.has("storyboards")) {
+            final JsonArray storyboardsArray = storyboards.getArray("storyboards");
+            for (final Object storyboard : storyboardsArray) {
+                if (storyboard instanceof JsonObject) {
+                    final JsonObject storyboardObject = (JsonObject) storyboard;
+                    final String url = storyboardObject.getString("storyboardPath");
+                    final int width = storyboardObject.getInt("spriteWidth");
+                    final int height = storyboardObject.getInt("spriteHeight");
+                    final int totalWidth = storyboardObject.getInt("totalWidth");
+                    final int totalHeight = storyboardObject.getInt("totalHeight");
+                    final int framesPerPageX = totalWidth / width;
+                    final int framesPerPageY = totalHeight / height;
+                    final int count = framesPerPageX * framesPerPageY;
+                    final int durationPerFrame = storyboardObject.getInt("spriteDuration") * 1000;
+
+                    framesets.add(new Frameset(
+                            List.of(baseUrl + url), // there is only one image per video
```

Comment:
One per video? in https://joinpeertube.org/news/release-6.0 it looks like there’s multiple per video

Reply:
Ah, it’s a composite image containing all of the thumbnails e.g. https://tube.tchncs.de/lazy-static/storyboards/8b7510d4-04bd-4937-90fd-2714c8cb4887.jpg